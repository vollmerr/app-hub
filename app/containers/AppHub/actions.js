/*
 *
 * AppHub actions
 *
 */

import {
  CHANGE_MOBILE,
} from './constants';

export function changeMobile(isMobile) {
  return {
    isMobile,
    type: CHANGE_MOBILE,
  };
}
