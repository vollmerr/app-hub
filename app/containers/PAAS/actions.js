import { createAction } from 'redux-actions';

import * as C from './constants';


export const getManagerDataRequest = createAction(C.GET_MANAGER_DATA_REQUEST);
export const getManagerDataSuccess = createAction(C.GET_MANAGER_DATA_SUCCESS);
export const getManagerDataFailure = createAction(C.GET_MANAGER_DATA_FAILURE);

export const getReportDataRequest = createAction(C.GET_REPORT_DATA_REQUEST);
export const getReportDataSuccess = createAction(C.GET_REPORT_DATA_SUCCESS);
export const getReportDataFailure = createAction(C.GET_REPORT_DATA_FAILURE);

export const updateUsersRequest = createAction(C.UPDATE_USERS_REQUEST);
export const updateUsersSuccess = createAction(C.UPDATE_USERS_SUCCESS);
export const updateUsersFailure = createAction(C.UPDATE_USERS_FAILURE);
