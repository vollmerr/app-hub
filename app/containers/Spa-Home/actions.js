/*
 *
 * SpaHome actions
 *
 */

import {
  EXAMPLE_DATA_REQUEST,
  EXAMPLE_DATA_SUCCESS,
  EXAMPLE_DATA_FAILURE,
} from './constants';

export function exampleDataRequest() {
  return {
    type: EXAMPLE_DATA_REQUEST,
  };
}

export function exampleDataSuccess(data) {
  return {
    data,
    type: EXAMPLE_DATA_SUCCESS,
  };
}

export function exampleDataFailure(error) {
  return {
    error,
    type: EXAMPLE_DATA_FAILURE,
  };
}
