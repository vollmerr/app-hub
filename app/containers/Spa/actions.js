import { createAction } from 'redux-actions';

import {
  INIT_DATA_REQUEST,
  INIT_DATA_SUCCESS,
  INIT_DATA_FAILURE,
  NEW_ACK_REQUEST,
  NEW_ACK_SUCCESS,
  NEW_ACK_FAILURE,
  DISABLE_ACK_REQUEST,
  DISABLE_ACK_SUCCESS,
  DISABLE_ACK_FAILURE,
  GET_RECIPIENTS_REQUEST,
  GET_RECIPIENTS_SUCCESS,
  GET_RECIPIENTS_FAILURE,
} from './constants';
// INITAL DATA
export const initDataRequest = createAction(INIT_DATA_REQUEST);
export const initDataSuccess = createAction(INIT_DATA_SUCCESS);
export const initDataFailure = createAction(INIT_DATA_FAILURE);
// NEW ACKNOWLEDGMENT
export const newAckRequest = createAction(NEW_ACK_REQUEST);
export const newAckSuccess = createAction(NEW_ACK_SUCCESS);
export const newAckFailure = createAction(NEW_ACK_FAILURE);
// DSIABLE ACKNOWLEDGMENT
export const disableAckRequest = createAction(DISABLE_ACK_REQUEST);
export const disableAckSuccess = createAction(DISABLE_ACK_SUCCESS);
export const disableAckFailure = createAction(DISABLE_ACK_FAILURE);
// GET LIST OF RECIPIENTS FOR EXISTSING ACK
export const getRecipientsRequest = createAction(GET_RECIPIENTS_REQUEST);
export const getRecipientsSuccess = createAction(GET_RECIPIENTS_SUCCESS);
export const getRecipientsFailure = createAction(GET_RECIPIENTS_FAILURE);
