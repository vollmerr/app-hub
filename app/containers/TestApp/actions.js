import { createActions } from 'redux-actions';

import {
  INIT_DATA_REQUEST,
  INIT_DATA_SUCCESS,
  INIT_DATA_FAILURE,
  CLEAR_ERRORS,
} from './constants';

const actions = createActions({
  [INIT_DATA_REQUEST]: () => {},
  [INIT_DATA_SUCCESS]: (payload) => payload,
  [INIT_DATA_FAILURE]: (payload) => payload,
  [CLEAR_ERRORS]: () => {},
});

export default actions.app.testApp;
