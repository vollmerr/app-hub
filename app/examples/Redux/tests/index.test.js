import React from 'react';
import { shallow } from 'enzyme';

import { Redux, mapDispatchToProps } from '../index';
import { exampleAction } from '../actions';
import Input from '../Input';


describe('<Redux />', () => {
  let wrapper;
  let mockFn;
  let mockData;

  beforeEach(() => {
    mockFn = jest.fn();
    mockData = 'test data';
    wrapper = shallow(
      <Redux exampleData={mockData} onExampleAction={mockFn} />
    );
  });

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a single input', () => {
    expect(wrapper.find(Input).length).toBe(1);
  });

  it('should handle entering text in the input', () => {
    wrapper.find(Input).dive().setProps({ onChange: mockFn }).simulate('change');
    expect(mockFn).toHaveBeenCalled();
  });

  it('should render data if it exists', () => {
    expect(wrapper.find('p').length).toBe(1);
  });

  it('should not render data if it does not exist', () => {
    wrapper.setProps({ exampleData: null });
    expect(wrapper.find('p').length).toBe(0);
  });

  describe('handleUpdateData', () => {
    it('should exist', () => {
      const instance = wrapper.instance();
      expect(instance.handleUpdateData).toBeDefined();
    });

    it('should dispatch onExampleAction when called', () => {
      const instance = wrapper.instance();
      const event = { target: { value: mockData } };
      instance.handleUpdateData(event);
      expect(mockFn).toHaveBeenCalled();
    });
  });

  describe('mapDispatchToProps', () => {
    let mappedDispatch;
    beforeEach(() => {
      mappedDispatch = mapDispatchToProps(mockFn);
    });

    describe('onExampleAction', () => {
      it('should be injected', () => {
        expect(mappedDispatch.onExampleAction).toBeDefined();
      });

      it('should dispatch exampleRequest when called', () => {
        mappedDispatch.onExampleAction();
        expect(mockFn).toHaveBeenCalledWith(exampleAction());
      });
    });
  });
});
