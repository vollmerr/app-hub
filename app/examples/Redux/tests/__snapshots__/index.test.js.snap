// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Redux /> should render correctly 1`] = `
<Example
  desc={
    Array [
      "This example is using redux.",
      "It is recommended to use the Redux Dev Tools for chrome to see the redux store change
    (both in this example and while developing...).",
      "Enter some text into the input field below. The text is updated from the redux store.
    Then try navigating to another example and back, notice the input's state does not persist,
    but the redux store's state does.",
      "Redux also allows us to easily access state accross different components, and
    should only be used in container components (ones that have logic, etc...)",
      "Redux should only be used in container components. When generating a new component
    (npm run generate) select 'y' for 'Do you want an actions/constants/selectors/reducer
    tuple for this container?'",
      "NOTE: Do not use redux in a container that has routing declared, it currently breaks it!",
    ]
  }
  header="Redux Example"
>
  <Input__Input
    onChange={[Function]}
    placeholder="Enter some text"
  />
  <p>
    test data
  </p>
</Example>
`;
